### ISSUE TITLE
WebClient documentation needs better and more examples

### ISSUE THREAD

**Issue by aoberoi** [slackapi/node-slack-sdk#366]
* [x] I've read and understood the [Contributing guidelines](./CONTRIBUTING.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](./CODE_OF_CONDUCT.md).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

#### Description

The WebClient should probably be the top-most example in the README. Also, we should show more than one usage of it (with only required arguments, with optional arguments, with a callback, with a Promise, with more than one method, without a token on initialization).

props to @shanedewael

**Comment by Rolias** [comment]
I would also like to cast a vote for some better examples. **EVEN MORE IMPORTANTLY** - In the docs on the Slack Developer Kit for Node.js in the [General Web API patterns](https://slackapi.github.io/node-slack-sdk/basic_usage#general-web-api-patterns)  There is an example showing how to handle optional parameters **HOWEVER** it would appear to be quite misleading (aka wrong). It shows a JSON object being passed in directly after the required objects but I found I had to JSON.stringify() any JSON object first. I couldn't find this documented anywhere other than [StackOverflow](https://stackoverflow.com/questions/35755165/slackbot-attachment-is-something-wrong-with-this). I was working with the webClient.users.profile.set() API and a good example here would have been hugely valuable.
